src/main/java/org/testng/Assert.java

```diff
@@ -1161,11 +1161,17 @@ public static void assertEquals(Object[] actual, Object[] expected, String messa
       Object a = actual[i];
       String explanation = "Arrays differ at element [" + i + "]: " + e + " != " + a;
       String errorMessage = message == null ? explanation : message + ": " + explanation;
-      //Compare by value for multi-dimensional array.
-      if(e.getClass().isArray()) {
-        assertEquals(a, e, errorMessage);
-      } else {
-        assertEqualsImpl(a, e, errorMessage);
+      if(a == null && e == null) continue;
+      if ((a == null && e != null) || (a != null && e == null)) {
+        failNotEquals(a, e, message);
+      }
+      if(a != null && e != null){
```

Comment:
I think we can do away with this condition because its already been taken care of by the other two edit checks above it.

Reply:
Sure, I agree.